File: frontend/src/lib/api/company.ts
import { MatchItem } from '../types';

type GenerateReasonsProps = {
  matchResult: MatchItem[];
  questions: {
    reasonInterest: string;
    attractiveService: string;
    relatedExperience: string;
  };
};

type PostReasonsProps = {
  content: string;
  companyName: string;
  companyUrl: string;
};

export const companyScrape = async (body: { companyUrl: string }) => {
  const url = 'http://localhost:8080/api/company-scrape';
  const token = localStorage.getItem('access_token');

  const data = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(body),
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    },
  });

  return data.json();
};

export const generateReasons = async (body: GenerateReasonsProps) => {
  const url = 'http://localhost:8080/api/generate-reasons';
  const token = localStorage.getItem('access_token');

  const data = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(body),
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    },
  });

  return data.json();
};

export const saveCompanyReason = async (props: PostReasonsProps) => {
  const url = 'http://localhost:8080/api/reason';
  const token = localStorage.getItem('access_token');

  const data = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(props),
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    },
  });

  return data.json();
};

export const getCompanyReasons = async () => {
  const token = localStorage.getItem('access_token');
  const res = await fetch('http://localhost:8080/api/reasons', {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  return res.json();
};

File: backend/internal/domain/model/company.go
package model

import "time"

type Reason struct {
	ID          uint   `gorm:"primaryKey;autoIncrement"`
	Content     string `gorm:"not null"`
	UserID      uint   `gorm:"not null"`
	User        User   `gorm:"foreignKey:UserID"`
	CompanyName string `gorm:"not null"`
	CompanyUrl  string `gorm:"not null"`
	CreatedAt   time.Time
}

type MatchItem struct {
	Axis   string `json:"axis"`
	Score  int    `json:"score"`
	Reason string `json:"reason"`
}

type GenerateReasonsRequest struct {
	MatchResult []MatchItem     `json:"matchResult"`
	Questions   QuestionAnswers `json:"questions"`
}

type QuestionAnswers struct {
	ReasonInterest    string `json:"reasonInterest"`
	AttractiveService string `json:"attractiveService"`
	RelatedExperience string `json:"relatedExperience"`
}
